genesis/utils/array_class.py

```diff
@@ -2004,6 +2005,70 @@ def __init__(self):
         return ClassEntitiesState()
 
 
+# =========================================== StaticRigidSimConfig ===========================================
+
+
+@dataclasses.dataclass
+class StaticRigidSimDummy:
+    para_level: int = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    use_hibernation: bool = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    use_contact_island: bool = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    batch_links_info: bool = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    batch_dofs_info: bool = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    batch_joints_info: bool = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    enable_mujoco_compatibility: bool = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    enable_multi_contact: bool = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    enable_self_collision: bool = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    enable_adjacent_collision: bool = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    enable_collision: bool = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    box_box_detection: bool = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    integrator: int = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    sparse_solve: bool = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    solver_type: int = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+
+    # dynamic properties
+    substep_dt: float = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    iterations: int = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    tolerance: float = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    ls_iterations: int = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    ls_tolerance: float = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    n_equalities: int = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    n_equalities_candidate: int = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    hibernation_thresh_acc: float = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
+    hibernation_thresh_vel: float = dataclasses.field(metadata={FIELD_METADATA_CACHE_VALUE: True})
```

Comment:
There was no request ot support floats, only ints and bools. Whilst it seems plausible tht it might work with floats, note that this is not tested. Please consider creating a PR onto gstaichi to add a test for floats. Something like https://github.com/Genesis-Embodied-AI/gstaichi/blob/a00b3aa5154b4b983df0c181cba3d60e20d190a8/tests/python/gstaichi/lang/fast_caching/test_args_hasher.py#L245 . (Yes, I agree that it also doesn't test bools, but floats seem more concerning to me, because of the potential for rounding issues).

Reply:
sure, technically, float doesn't need to be `FIELD_METADATA_CACHE_VALUE` since they are not used in ti.static. I will move those float values out of this class.